modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/utils/dataConverter.es.js

```diff
@@ -0,0 +1,370 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {FieldSupport} from 'dynamic-data-mapping-form-builder';
+import {PagesVisitor} from 'dynamic-data-mapping-form-renderer';
+
+import {getDataDefinitionField as getDataDefinitionFieldUtils} from './dataDefinition.es';
+import {
+	normalizeDataDefinition,
+	normalizeDataLayout,
+	normalizeDataLayoutRows,
+} from './normalizers.es';
+
+export function getDataDefinitionFieldSet({
+	allowInvalidAvailableLocalesForProperty,
+	availableLanguageIds,
+	defaultLanguageId,
+	editingLanguageId,
+	fieldSet,
+	fieldTypes,
+}) {
+	const dataLayoutPages = (
+		fieldSet.defaultDataLayout || getDefaultDataLayout(fieldSet)
+	).dataLayoutPages;
```

Comment:
```suggestion
	const fieldSetDataLayout = fieldSet.defaultDataLayout || getDefaultDataLayout(fieldSet);
	const fieldSetDataLayoutPages = fieldSetDataLayout.dataLayoutPages;
```

and then replace `dataLayoutPages` usages with `fieldSetDataLayoutPages`. What do you think?